wasmtime-runtime/src/jit_dump.rs

```diff
@@ -0,0 +1,701 @@
+//! Support for jitdump files which can be used by perf for profiling jitted code.
+//! Spec definitions for the output format is as described here:
+//! https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/perf/Documentation/jitdump-specification.txt
+//!
+//! Usage Example:
+//!     Record
+//!         sudo perf record -k 1 -e instructions:u target/debug/wasmtime -g -jitdump test.wasm
+//!     Combine
+//!         sudo perf inject -v -j -i perf.data -o perf.jit.data
+//!     Report
+//!         sudo perf report -i perf.jit.data -F+period,srcline
+//! Note: For descriptive results, the WASM file being executed should contain dwarf debug data
+
+use libc::{c_int, c_void, clockid_t, mmap, open, sysconf, timespec};
+use object::Object;
+use scroll::{IOwrite, SizeWith, NATIVE};
+use serde::{Deserialize, Serialize};
+use std::ffi::CString;
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::Write;
+use std::os::unix::io::FromRawFd;
+use std::{borrow, mem, process};
+use target_lexicon::Architecture;
+
+#[cfg(target_pointer_width = "64")]
+use goblin::elf64 as elf;
+
+#[cfg(target_pointer_width = "32")]
+use goblin::elf32 as elf;
+
+/*
+ * The following record types are defined:
+ * Value 0 : JIT_CODE_LOAD: record describing a jitted function
+ * Value 1 : JIT_CODE_MOVE: record describing an already jitted function which is moved
+ * Value 2 : JIT_CODE_DEBUG_INFO: record describing the debug information for a jitted function
+ * Value 3 : JIT_CODE_CLOSE: record marking the end of the jit runtime (optional)
+ * Value 4 : JIT_CODE_UNWINDING_INFO: record describing a function unwinding information
+ */
```

Comment:
Please use doc comments instead. https://doc.rust-lang.org/rust-by-example/meta/doc.html

Also please move the variant docs to the variants themself.

```rust
/// The type of a record
enum RecordId {
    /// Record describing a jitted function
    JitCodeLoad = 0,
    // ...
}